test/python/common.py

```diff
@@ -312,46 +354,34 @@ def requires_qe_access(func):
 
     @functools.wraps(func)
     def _(*args, **kwargs):
-        # pylint: disable=invalid-name
-        if SKIP_ONLINE_TESTS:
-            raise unittest.SkipTest('Skipping online tests')
-
         # Cleanup the credentials, as this file is shared by the tests.
         from qiskit.wrapper import _wrapper
         _wrapper._DEFAULT_PROVIDER = DefaultQISKitProvider()
 
-        if os.getenv('USE_ALTERNATE_ENV_CREDENTIALS', False):
-            # Special case: instead of using the standard credentials mechanism,
-            # load them from different environment variables. This assumes they
-            # will always be in place, as is used by the Travis setup.
+        if TEST_OPTIONS['skip_online']:
+            raise unittest.SkipTest('Skipping online tests')
+
+        if get_credentials(args, kwargs):
+            if TEST_OPTIONS['rec'] or TEST_OPTIONS['mock_online']:
```

Comment:
if you are recording (`TEST_OPTIONS['rec']`) or you are mocking (`TEST_OPTIONS['mock_online']`), you need to decorate the test function. Otherwise, you go as currently.